fedimint-server/src/config/api.rs

```diff
@@ -384,6 +434,99 @@ impl ConfigGenApi {
     fn bad_request<T>(msg: &str) -> ApiResult<T> {
         Err(ApiError::bad_request(msg.to_string()))
     }
+
+    pub async fn restart_federation_setup(&self) -> ApiResult<()> {
+        let leader = {
+            let expected_status = [
+                ServerStatus::SharingConfigGenParams,
+                ServerStatus::ReadyForConfigGen,
+                ServerStatus::ConfigGenFailed,
+                ServerStatus::VerifyingConfigs,
+                ServerStatus::VerifiedConfigs,
+            ];
+            let mut state = self.require_any_status(&expected_status)?;
+
+            let cfg_staging_dir = self.data_dir.join(CONFIG_STAGING_DIR);
+            self.remove_staged_configs(&cfg_staging_dir)?;
+
+            state.status = ServerStatus::SetupRestarted;
+            info!(
+                target: fedimint_logging::LOG_NET_PEER_DKG,
+                "Update config gen status to 'Setup restarted'"
+            );
+            // Create a WSClient for the leader
+            state
+                .local
+                .clone()
+                .and_then(|local| local.leader_api_url.map(WsAdminClient::new))
+        };
```

Comment:
What's the reason to put this into a separate block?

Reply:
this allows us to acquire lock on `self.state`, use it and and releasing lock, before we call `update_leader`. update leader creates a separate lock on state

Reply:
I didn't remember that `require_any_status` returned a mutex guard, makes sense!